### PR TITLE
Helm chart improvements

### PR DESCRIPTION
**Pull request by inteon**
- remove deprecated `storage.k8s.io/v1beta1`
- add namespace values that make sure namespace is included when running `helm template`
- use dynamic `$(POD_NAMESPACE)` instead of static templated `{{ .Release.Namespace }}` value (should resolve to same value)

### CODE REVIEW

**Comment by inteon** [comment]
> The PR looks good to me, except for I'd like to understand the effect of [this change](https://github.com/cert-manager/csi-driver-spiffe/pull/37/commits/ef1e2f8600272b95acf0be6e1b9af8e26ef1be3b#diff-70494d85431935e8587be62cfb49aacdf42ad2ead54c85794cd9218acf398834R55) if a user upgrades the csi-driver-spiffe to this version and already has some csi-driver-spiffe volumes

I removed that change, so we can easily merge this PR.

File: deploy/charts/csi-driver-spiffe/templates/deployment.yaml [hunk 1]
```diff
@@ -37,9 +38,14 @@ spec:
...
 
-          - --leader-election-namespace={{ .Release.Namespace }}
+          - --leader-election-namespace=$(POD_NAMESPACE)
           - "--metrics-bind-address=:{{.Values.app.approver.metrics.port}}"
           - "--readiness-probe-bind-address=:{{.Values.app.approver.readinessProbe.port}}"
+        env:
+        - name: POD_NAMESPACE
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.namespace
```
Comment (SgtCoDFish) [inline deploy/charts/csi-driver-spiffe/templates/deployment.yaml:48]: question: What's the motivation for this change? I'm not saying we shouldn't do it - I just don't know why we _are_ doing it

**Review by SgtCoDFish — APPROVED** [review]
/lgtm
/approve
/hold

Looks good! Nothing here to block merging, but adding a hold for my question. Feel free to unhold!

File: deploy/charts/csi-driver-spiffe/templates/deployment.yaml (hunk 1 above)
Comment (inteon) [reply deploy/charts/csi-driver-spiffe/templates/deployment.yaml:48]: This is a very subjective change, it makes it so our operator manifests have less namespaces that have to be updated when we change the namespace. This change is copied from the changes we had in that repo.
